server.js

```diff
@@ -347,6 +347,54 @@ cache(function(data, match, sendBadge, request) {
   });
 }));
 
+camp.route(/^\/crates\/(d|v|dv)\/([A-Za-z0-9_-]+)(?:\/([0-9.]+))?\.(svg|png|gif|jpg|json)$/,
+cache(function (data, match, sendBadge, request) {
+  var mode = match[1];
+  var crate = match[2];
+  var version = match[3];
+  var format = match[4];
+  var apiUrl = 'https://crates.io/api/v1/crates/' + crate;
+  if (version != null) {
+    apiUrl += '/' + version;
+  }
+  var badgeData = getBadgeData(mode == 'v'? 'version': 'downloads', data);
+  request(apiUrl, { headers: { 'Accept': 'application/json' } }, function (err, res, buffer) {
+    if (err != null) {
+      badgeData.text[1] = 'inaccessible';
+      sendBadge(badgeData, format);
+    }
+    try {
+      var data = JSON.parse(buffer);
+      var downloads;
+
+      if (mode == 'd') {
+        downloads = data.crate.downloads || data.version.downloads;
+        version = data.version.num;
```

Comment:
When no version is provided, `data` seems to only contain keys `crate`, `keywords`, `versions`. This code does not work.